### DISCUSSION TITLE
How to npm link packages from turborepo to external consumer app

### DISCUSSION THREAD

**Discussion by ghost** [vercel/turborepo#2864] [Help] [answered]
Hi, I have a setup where I'm using Turborepo for a component library. Structure as such:

```
/app // preview app
  ...
/packages
  /core
    ...
  /example-component
    // has dependencies
    // core: "*"
    // @styled/emotion
    ...
```

I then have a consumer app which isn't part of Turborepo. It has `example-component` as a dependency. When `example-component` is published, I can install and all is fine.

But whilst developing I want to npm link this package but come across some issues. First issue is:

- Requires me to have `@styled/emotion` installed - fine. I install this. no problem

Then

- Requires me to have to have `core installed`. I've tred adding `core` as a dependency to the consumer app and then `npm link` this, but then `example-component` link seems to be removed

Also, as `core` is a dependency of `example-component` and I'd would like any changes made here reflected in `example-component` how would the linking work 

What is the workflow for this kind of scenario?

**Comment by anthonyshew** [comment, marked as answer]
You may want to learn more about `peerDependencies` from the Node.js spec!

https://nodejs.org/en/blog/npm/peer-dependencies
